File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
     let arr = nums.map((value, index) =>({
        value,
        index
    }));
    arr.sort((a,b) => a.value - b.value)

     let left = 0
     let right = arr.length-1
   
   while(left<right){
    let sum = arr[left].value + arr[right].value
    if(sum === target){
        return [arr[left].index, arr[right].index]
    }else if (sum<target){
 left++
    }
    else{
        right--
    }
   }
    return []
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        Map<Integer,Integer> map = new HashMap<>();
        
        for(int i=0; i<n; i++){
            int rem = target - nums[i];

            if(map.containsKey(rem)){
                return new int[] {map.get(rem),i};
            }
            else{
                map.put(nums[i],i);
            }
        }
        return new int[]{};

    }
}